engine/table/src/main/java/io/deephaven/engine/util/PyCallableWrapper.java

```diff
@@ -0,0 +1,266 @@
+        this.pyCallable = pyCallable;
+    }
+
+    public ArgumentsChunked buildArgumentsChunked(List<String> columnSources) {
+        Object[] args;
+        Class<?>[] argTypes;
+
+        if (!numbaVectorized) {
+            args = new Object[chunkArguments.size() + 1];
+            argTypes = new Class[chunkArgumentTypes.length + 1];
+            argTypes[0] = int.class;
+        } else {
+            args = new Object[chunkArguments.size()];
+            argTypes = new Class[chunkArgumentTypes.length];
+        }
+
+        int i = numbaVectorized ? 0 : 1;
+        int j = 0;
+        for (ChunkArgument arg : chunkArguments) {
+            if (arg instanceof ColumnChunkArgument) {
+                String columnName = ((ColumnChunkArgument) arg).getColumnName();
+                int chunkSourceIndex = columnSources.indexOf(columnName);
+                if (chunkSourceIndex < 0) {
+                    throw new IllegalArgumentException("Column source not found: " + columnName);
+                }
+                ((ColumnChunkArgument) arg).setChunkSourceIndex(chunkSourceIndex);
+            } else {
+                args[i] = ((ConstantChunkArgument) arg).getValue();
+                argTypes[i] = chunkArgumentTypes[j];
+            }
+            i++;
+            j++;
+        }
+        return new ArgumentsChunked(chunkArguments, args, argTypes, numbaVectorized);
+    }
+
+    public interface ChunkArgument {
+    }
+    public static class ColumnChunkArgument implements ChunkArgument {
+        private final String columnName;
+        private int chunkSourceIndex;
+        private boolean resolved = false;
+
+        public ColumnChunkArgument(String columnName) {
+            this.columnName = columnName;
+        }
+
+        public void setChunkSourceIndex(int chunkSourceIndex) {
+            this.resolved = true;
+            this.chunkSourceIndex = chunkSourceIndex;
+        }
+
+        public int getChunkSourceIndex() {
+            if (!resolved) {
+                throw new IllegalStateException("The column chunk argument hasn't been resolved " + this.columnName);
+            }
+            return chunkSourceIndex;
+        }
+
+        public String getColumnName() {
+            return columnName;
+        }
+    }
+
+    public static class ConstantChunkArgument implements ChunkArgument {
+        private final Object value;
+
+        public ConstantChunkArgument(Object value) {
+            this.value = value;
+        }
+
+        public Object getValue() {
+            return value;
+        }
+    }
+
+    // this assumes that the Python interpreter won't be re-initialized during a session, if this turns out to be a
+    // false assumption, then we'll need to make this initialization code 'python restart' proof.
+    static PyObject getNumbaVectorizedFuncType() {
+        try {
+            return PyModule.importModule("numba.np.ufunc.dufunc").getAttribute("DUFunc");
```

Comment:
Do we also need to support `ufuncs`?  
https://numba.pydata.org/numba-doc/latest/reference/jit-compilation.html#vectorized-functions-ufuncs-and-dufuncs

Reply:
My guess is no, though I found the doc not very clear.